compiler/rustc_trait_selection/src/solve/canonical/canonicalize.rs

```diff
@@ -0,0 +1,384 @@
+use std::cmp::Ordering;
```

Comment:
Noting from our discussion that this probably will need to live in infer after all.